p32/src/bank.rs

```diff
@@ -0,0 +1,104 @@
+pub enum Error {
+    AmountTooLarge,
+    SenderNotFound,
+    ReceiverNotFound,
+    CreditLimitOverflow,
+    BadCreditLimit,
+}
+pub struct User {
+    name: String,
+    credit_line: u64,
+    balance: i64,
+}
+pub struct Bank {
+    users: Vec<User>,
+    name: String,
+    credit_interest: u64,
+    debit_interest: u64,
+}
+
+impl Bank {
+    pub fn calc_balance(&self) -> (u64, u64) {
+        let mut liab: u64 = 0;
+        let mut ass: u64 = 0;
+        for i in self.users.iter() {
+            if i.balance < 0 {
+                ass += i.balance.unsigned_abs();
+            } else {
+                liab += i.balance.unsigned_abs();
+            }
+        }
+        (liab, ass)
+    }
+
+    pub fn transfer_funds(
+        &mut self,
+        user1: String,
+        user2: String,
+        amount: u64,
+    ) -> Result<bool, Error> {
```

Comment:
`bool` does not contain any useful information here. You should return `Result<(), Error>`.